src/App.tsx

```diff
@@ -38,58 +39,124 @@ export function getReorderedGoods(
+      });
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((currentGood, nextGood) => {
+        return currentGood.length - nextGood.length;
+      });
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByAlphabetis = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => {
+      return {
+        isReversed: !state.isReversed,
+      };
+    });
+  };
+
+  resetAllSorts = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              { 'is-light': this.state.sortType !== SortType.ALPHABET },
+            )}
+            onClick={this.sortByAlphabetis}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': this.state.sortType !== SortType.LENGTH },
+            )}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning',
+              { 'is-light': !this.state.isReversed },
+            )}
+            onClick={this.reverseGoods}
+          >
+            Reverse
+          </button>
+
+          {(this.state.isReversed !== false
+          || this.state.sortType !== SortType.NONE)
```

Comment:
the condition better like this:
`(sortType !== SortType.NONE || isReversed)`

than this:
`this.state.isReversed !== false`
really strange 😅

Reply:
It could be extremely strange, but it is my decision. You should be more polite with your words, cause your words to sound a bit offensive

Reply:
i am so sorry i didn't want to offend you (((

Reply:
Ok, np)